graphql/query_resolver.go

```diff
@@ -121,7 +121,17 @@ func (r *queryResolver) Distro(ctx context.Context, distroID string) (*restModel
 
 // DistroEvents is the resolver for the distroEvents field.
 func (r *queryResolver) DistroEvents(ctx context.Context, opts DistroEventsInput) (*DistroEventsPayload, error) {
-	events, err := event.FindLatestPrimaryDistroEvents(opts.DistroID, utility.FromIntPtr(opts.Limit), opts.Before)
+	before := time.Now()
+	if opts.Before != nil {
+		before = *opts.Before
```

Comment:
That makes sense. I also updated a lot of the usages of `FindLatestPrimaryDistroEvents` to pass in `utility.ZeroTime` instead of `time.Now()` to avoid the extra clause. Both should work though (and both are tested).